app/javascript/serviceworker.js

```diff
@@ -0,0 +1,26 @@
+import './serviceworker'
+
+if ('serviceWorker' in navigator && navigator !== null) {
+  navigator.serviceWorker.register('/serviceworker.js')
+}
+
+self.addEventListener('install', (event) => {
+  // Perform install steps
+  event.waitUntil(
+    caches.open('static').then(function (cache) {
+      return cache.addAll(['/', '/index.html', '/css/style.css', '/js/app.js'])
```

Comment:
@jasonpaulso I'm not super familiar with service workers, but it looks like this is caching some specific files. Will these files be properly reloaded when new versions are pushed? I'm noticing those don't have the usual rails asset pipeline cache busting stuff within the filenames.

Reply:
Apologies for the confusion with that file—the service worker stuff is somewhat new to me as well, in particular with how it works in Rails. I got a bit overzealous and ultimately forgot to revisit that part when before I submitted the pull request. My mistake! It is safe to disregard that part for now. It's not strictly necessary (and isn't really doing anything yet). I should have time to properly flesh that out tomorrow.